### PR TITLE
fix: backend returns 204 against Camel Route JSON with script step

### LINKED ISSUE #529: [BUG] backend returns 204 against Camel Route JSON with script step
**Describe the bug**
The backend returns 204 against the Camel Route JSON with script step.

https://user-images.githubusercontent.com/265462/224376501-6b7b67a1-fb1f-452b-bd51-bba8f97b7d6b.mp4

**To Reproduce**
1. Add activemq step
2. Add Script step
3. The source code editor loses the source

**Expected behavior**
Code editor should stay showing the code

**Logs**
Captured request body JSON
[script-204.json.txt](https://github.com/KaotoIO/kaoto-backend/files/10944327/script-204.json.txt)

### PR DESCRIPTION
**Pull request by igarashitm**
Fixes: #529

https://github.com/KaotoIO/camel-component-metadata/pull/12 needs to go first

### CODE REVIEW

**Comment by igarashitm** [comment]
Screenshot from 2023-03-16 17-19-26

File: api/src/test/resources/io/kaoto/backend/api/resource/eip.kamelet.yaml [hunk 1]
```diff
@@ -45,8 +45,7 @@ spec:
                   sample-period: '1500'
                   message-frequency: 5
               - script:
-                  expression:
```
Comment (Delawen) [inline api/src/test/resources/io/kaoto/backend/api/resource/eip.kamelet.yaml:48]: Script should support expression, why remove it?

 https://github.com/apache/camel/blob/main/dsl/camel-yaml-dsl/camel-yaml-dsl/src/test/groovy/org/apache/camel/dsl/yaml/ScriptTest.groovy

Comment (igarashitm) [reply api/src/test/resources/io/kaoto/backend/api/resource/eip.kamelet.yaml:48]: We can add like `ScriptExpression` and do the same thing with `Expression` but also keep script expression isolated from other expression. but...

It reminded me that in the past, you said we need to support it from step extensions as well, and I haven't worked on it yet. Is it considered as a bug? Is there a issue for that? It'd need to be created or added to Sprint 16 I guess.

**Comment by Delawen** [comment]
Can you check the failed tests?

**Comment by igarashitm** [comment]
@Delawen tests should pass once https://github.com/KaotoIO/camel-component-metadata/pull/12 is merged I guess?

**Comment by Delawen** [comment]
Then let's do it!

**Review by Delawen — APPROVED** [review]

**Comment by Delawen** [comment]
Aah I guess we need to merge this before tests in other PR pass again. We need versioning on the camel components repo.

**Comment by igarashitm** [comment]
ugh OK, I'll address https://github.com/KaotoIO/kaoto-backend/issues/547 in a separate PR then, with a separate PR to component metadata as well...

**Pull request merged**